buildenv/jenkins/jobs/infrastructure/artifactoryCheck.groovy

```diff
@@ -0,0 +1,67 @@
+ * Exception [1] and GNU General Public License, version 2 with the
+ * OpenJDK Assembly Exception [2].
+ *
+ * [1] https://www.gnu.org/software/classpath/license.html
+ * [2] http://openjdk.java.net/legal/assembly-exception.html
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0 WITH Classpath-exception-2.0 OR LicenseRef-GPL-2.0 WITH Assembly-exception
+ *******************************************************************************/
+def LABEL = (params.LABEL != "") ? params.LABEL : "artifactory"
+def WARNING_THRESHOLD = (params.WARNING_THRESHOLD != "") ? params.WARNING_THRESHOLD : "75"
+def DANGER_THRESHOLD = (params.DANGER_THRESHOLD != "") ? params.DANGER_THRESHOLD : "90"
+def SLACK_CHANNEL = (params.SLACK_CHANNEL != "") ? params.SLACK_CHANNEL : "#jenkins-sandbox"
+HIGH_ROOT_DISK_PERCENTAGE = 0
+HIGH_ARTIFACTORY_DISK_PERCENTAGE = 0
+
+def artifactoryNodes = nodesByLabel label:"${LABEL}", offline: false
+
+timestamps {
+	echo "Number of Nodes that match labels: " + artifactoryNodes.size()
+	for (sNode in artifactoryNodes) {
+		node("${sNode}") {
+			stage("${sNode} Artifactory data collection\n") {
+				ROOT_DISK_DETAILS = sh (label: 'Get / disk details', returnStdout: true, script: "df / -h | tail -1 | awk '{print \$3 \" / \" \$2}'").trim()
+				ROOT_DISK_PERCENTAGE = sh (label: 'Get / disk percentage', returnStdout: true, script: "df / -h | tail -1 | awk '{print \$5}'").trim()
+				ARTIFACTORY_DISK_DETAILS = sh (label: 'Get Artifactory disk details', returnStdout: true, script: "df -h | grep jenkins | awk '{print \$3 \" / \" \$2}'").trim()
+				ARTIFACTORY_DISK_PERCENTAGE = sh (label: 'Get Artifactory disk percentage', returnStdout: true, script: "df -h | grep jenkins | awk '{print \$5}'").trim()
+				MESSAGE = "$Artifactory on ${sNode}:\nRoot disk used: ${ROOT_DISK_DETAILS} -> ${ROOT_DISK_PERCENTAGE}\nData disk used: ${ARTIFACTORY_DISK_DETAILS} -> ${ARTIFACTORY_DISK_PERCENTAGE}"
+                		echo "${MESSAGE}"
+
+				if (HIGH_ROOT_DISK_PERCENTAGE < (ROOT_DISK_PERCENTAGE - '%').toInteger()) {
+					HIGH_ROOT_DISK_PERCENTAGE = (ROOT_DISK_PERCENTAGE - '%').toInteger()
+					highNodeRoot = sNode
+				}
+				if (HIGH_ARTIFACTORY_DISK_PERCENTAGE < (ARTIFACTORY_DISK_PERCENTAGE - '%').toInteger()) {
+					HIGH_ARTIFACTORY_DISK_PERCENTAGE = (ARTIFACTORY_DISK_PERCENTAGE - '%').toInteger()
+					highNodeArtifactory = sNode
+				}
+			}
+		}
+	}
+
+	stage("Artifactory data review") {        
+		SLACK_COLOUR = "good"
+		if ((HIGH_ROOT_DISK_PERCENTAGE > DANGER_THRESHOLD) || (HIGH_ARTIFACTORY_DISK_PERCENTAGE > DANGER_THRESHOLD)) {
+			SLACK_COLOUR = "danger"
+		}
+		if ((HIGH_ROOT_DISK_PERCENTAGE > WARNING_THRESHOLD) || (HIGH_ARTIFACTORY_DISK_PERCENTAGE > WARNING_THRESHOLD)) {
+			SLACK_COLOUR = "warning"
+		}
+		MESSAGE = "Highest root disk percentage ${HIGH_ROOT_DISK_PERCENTAGE}% on ${highNodeRoot}\nHighest artifactory disk percentage ${HIGH_ARTIFACTORY_DISK_PERCENTAGE}% on ${highNodeArtifactory}\nSlack colour ${SLACK_COLOUR}"
```

Comment:
Did we decide if we were going to echo/slack out the Geo instead of the node name? "worker-1" isn't as obvious as "OSU".

Reply:
I've modified and included both node and geo.